### DISCUSSION TITLE
How can I solve Not Found "*/java agent/v5/initConfig"

### DISCUSSION THREAD

**Discussion by sgh002400** [naver/scavenger#66] [Q&A] [answered]
I watched NAVER DEVIEW 2023 and admired the wonderful open source 😊. However, I encountered an error while attempting to analyze Dead Code through Scavenger. I would really appreciate it if you could help me solve it.

### 1. Placed in same location.
<img width="211" alt="image" src="https://user-images.githubusercontent.com/62228195/233790062-c40daff8-f449-4ba3-a00c-0b51bd162ace.png">

</br>

### 2. API
```java
java -Dscavenger.collector-server-url=http://localhost:9090 -jar scavenger-api-boot-1.0.4.jar
```
![C169590B-B511-40AD-A35A-B14BCBBF3111](https://user-images.githubusercontent.com/62228195/233790191-75d90be3-4508-4d53-9a84-640056079e5d.png)

</br>

### 3. Collector
```java
java -Darmeria.port=8080  -jar scavenger-collector-boot-1.0.4.jar
```
![DA26582E-0B15-4E01-87C8-91C3649B3568](https://user-images.githubusercontent.com/62228195/233790202-6043fb33-3efc-4f85-9d25-1d122272a4cb.png)
![B0BDF482-7844-48BF-B7DD-83151724E837](https://user-images.githubusercontent.com/62228195/233790206-51459e68-5f62-4b4f-afaf-15c5e05dffca.png)
![667FF1D8-40C7-4CCC-BC5E-6044996449CB](https://user-images.githubusercontent.com/62228195/233790210-b7fd0ace-5900-4c8d-a94e-401ff27a7429.png)

</br>

### 4. conf
- The scavenger.conf file was created here by http://localhost:8081/scavenger/customers/ratrip, and only the port was matched with the ratrip application.
![image](https://user-images.githubusercontent.com/62228195/233790545-1f38236b-3452-448d-964c-8524a5d9a55e.png)

</br>

### 5. Ratrip
```java
APPLE_CLIENT_ID=***** \
APPLE_KEY_ID=***** \
APPLE_REDIRECT_URI=***** \
APPLE_TEAM_ID=***** \
DB_HOST=***** \
DB_PASSWORD=***** \
DB_PORT=***** \
DB_SCHEMA=***** \
DB_USERNAME=***** \
JWT_SECRET_KEY=*****= \
KAKAO_API_KEY=***** \
KAKAO_REDIRECT_URI=***** \
NAVER_CLIENT_KEY=***** \
NAVER_SECRET_KEY=***** \
REDIS_HOST=***** \
REDIS_PORT=***** \
SPRING_PROFILES_ACTIVE=dev \
java -Dscavenger.configuration=scavenger.conf -javaagent:scavenger-agent-java-1.0.4.jar -jar ratrip-0.0.1-SNAPSHOT.jar
```
![77462778-A9AC-4791-B4C8-18AA6BA3178C](https://user-images.githubusercontent.com/62228195/233790300-a04504ef-828c-45e3-b163-83566573620b.png)
![AA6EA557-FDB8-4313-8B57-19E706CA3C73](https://user-images.githubusercontent.com/62228195/233790304-4e993d19-152e-43e6-9aa7-f9f9e3c43492.png)
![4FCD7F17-7514-45A7-9869-CFC154CD3EE4](https://user-images.githubusercontent.com/62228195/233790307-073515e0-3556-46f3-a501-acc7d2028ec6.png)

</br>

### Problem
` "status": 404, "error": "Not Found", "path": "/javaagent /v5/initConfig"} `
I wonder how to solve the problem. Thank you in advance for your reply 🙏

**Comment by kojandy** [comment, marked as answer]
Thank you for your interest in Scavenger!
You need to point your `serverUrl` to the Scavenger collector, not your application.
```
serverUrl = http://localhost:8080
```
Please double check your configuration and feel free to ask any following questions!

**Reply by taeyeon-Kim** [reply]
also `UNAUTHENTICATED` seems to be related to https://github.com/naver/scavenger/issues/23.
Can you check it out?

**Reply by sgh002400** [reply]
I appreciate the help from both of you 🙏. Thanks to your assistance, I was able to successfully resolve my issue and had a pleasant experience analyzing and cleaning up dead code. Thank you for developing such an amazing open source !
